tools/extra/hssi/hssicommon.py

```diff
@@ -0,0 +1,818 @@
+        return self.bits.port_read_latency
+
+    @property
+    def port_databus_width(self):
+        return self.bits.port_databus_width
+
+    @property
+    def low_speed_eth(self):
+        return self.bits.low_speed_eth
+
+    @property
+    def dyn_reconf(self):
+        return self.bits.dyn_reconf
+
+
+class hssi_cmd_sts_bits(Structure):
+    """
+    HSSI Feature Command/Status CSR bits
+    """
+    _fields_ = [
+                   ("read", c_uint32, 1),
+                   ("write", c_uint32, 1),
+                   ("ack", c_uint32, 1),
+                   ("busy", c_uint32, 1),
+                   ("error", c_uint32, 1),
+                   ("reserved", c_uint32, 27)
+    ]
+
+
+class hssi_cmd_sts(Union):
+    """
+    HSSI feature Feature Command/Status
+    Byte Offset: 0x50
+    Addressing Mode: 32 bits
+    """
+    _fields_ = [("bits", hssi_cmd_sts_bits),
+                ("value", c_uint32)]
+
+    def __init__(self, value):
+        self.value = value
+
+    @property
+    def ack(self):
+        return self.bits.ack
+
+    @property
+    def busy(self):
+        return self.bits.busy
+
+    @property
+    def error(self):
+        return self.bits.error
+
+    def read(self, value):
+        self.bits.read = value
+
+    def write(self, value):
+        self.bits.write = value
+
+    def ack(self, value):
+        self.bits.ack = value
+
+
+class HSSI_SALCMD(Enum):
+    """
+    HSSI Feature Control/Address Enum
+    """
+    NOP = 0
+    GET_HSSI_PROFILE = 0x1
+    SET_HSSI_PROFILE = 0x2
+    READ_MAC_STATISTIC = 0x3
+    GET_MTU = 0x4
+    SET_SCR = 0x5
+    GET_SCR = 0x6
+    ENABLE_LOOPBACK = 0x7
+    DISABLE_LOOPBACK = 0x8
+    FIRMWARE_VER = 0xFF
+
+
+class hssi_ctl_addr_bits(Structure):
+    """
+    HSSI Feature Control/Address CSR bits
+    """
+    _fields_ = [
+                   ("sal_cmd", c_uint32, 8),
+                   ("port_address", c_uint32, 4),
+                   ("ch_address", c_uint32, 4),
+                   ("addressbit", c_uint32, 16)
+    ]
+
+
+class hssi_ctl_addr(Union):
+    """
+    HSSI Feature Control/Address CSR
+    Byte Offset: 0x54
+    Addressing Mode: 32 bits
+    """
+    _fields_ = [("bits", hssi_ctl_addr_bits),
+                ("value", c_uint32)]
+
+    def __init__(self, value):
+        self.value = value
+
+    @property
+    def sal_cmd(self):
+        return self.bits.sal_cmd
+
+    @property
+    def port_address(self):
+        return self.bits.port_address
+
+    @property
+    def ch_address(self):
+        return self.bits.ch_address
+
+    @property
+    def addressbit(self):
+        return self.bits.addressbit
+
+    def addressbit(self):
+        self.bits.addressbit = value
+
+    def sal_cmd(self, value):
+        self.bits.sal_cmd = value
+
+    def port_address(self, value):
+        self.bits.port_address = value
+
+    def ch_address(self, value):
+        self.bits.ch_address = value
+
+    def addressbit(self, value):
+        self.bits.addressbit = value
```

Comment:
Should probably make these property setters for clarity.
```python
@sal_cmd.setter
def sal_cmd(self, value):
    self.bits.sal_cmd = value
```

Reply:
added setter functions , fixed

Reply:
Were you going to use the property setter syntax?  Instead of..
```python
 def set_sal_cmd(self, value):
        self.bits.sal_cmd = value
```
you could use..
```python
@sal_cmd.setter
def sal_cmd(self, value):
    self.bits.sal_cmd = value
```
Then, in the client code you can use..
```python
obj.sal_cmd = value
```

Reply:
fixed